api/features/views.py

```diff
@@ -187,6 +189,53 @@ def get_serializer_context(self):
 
         return context
 
+    def apply_state_to_queryset(
+        self, query_data: dict[str, typing.Any], queryset: QuerySet[Feature]
+    ) -> QuerySet[Feature]:
+        if not query_data.get("environment"):
+            raise serializers.ValidationError(
+                "Environment is required in order to filter by state search or by state enabled"
+            )
+        state_enabled = query_data["state_enabled"]
+        state_search = query_data["state_search"]
+        environment_id = query_data["environment"]
+
+        filter_search_q = Q()
+        if state_search is not None:
+            filter_search_q = filter_search_q | Q(
+                feature_state_value__string_value__icontains=state_search,
+            )
+
+            if state_search.lower() in {"true", "false"}:
+                boolean_search = state_search.lower() == "true"
+                filter_search_q = filter_search_q | Q(
+                    feature_state_value__boolean_value=boolean_search
+                )
+
+            if state_search.isdigit():
+                integer_search = int(state_search)
+                filter_search_q = filter_search_q | Q(
+                    feature_state_value__integer_value=integer_search
+                )
+        filter_enabled_q = Q()
+        if state_enabled is not None:
+            filter_enabled_q = filter_enabled_q | Q(enabled=state_enabled)
+
+        base_q = Q(
+            identity__isnull=True,
+            feature_segment__isnull=True,
+        )
+        if not getattr(self, "environment", None):
+            self.environment = Environment.objects.get(id=environment_id)
```

Comment:
I'm not sure what scenarios the environment is / isn't set on the instance already?

Reply:
If I recall correctly, this was done this way in order to play nicely with another PR that also needed to set the environment variable and it wasn't clear which of the methods would run first.